test/battle/ability/clear_body.c

```diff
@@ -26,13 +28,241 @@ SINGLE_BATTLE_TEST("Clear Body prevents intimidate")
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent Topsy-Turvy");
-TO_DO_BATTLE_TEST("Clear Body doesn't prevent Spectral Thief from resetting positive stat changes");
-TO_DO_BATTLE_TEST("Clear Body is ignored by Mold Breaker");
+SINGLE_BATTLE_TEST("Clear Body prevents stat stage reduction from moves")
+{
+    u16 move;
+    PARAMETRIZE{ move = MOVE_GROWL; }
+    PARAMETRIZE{ move = MOVE_LEER; }
+    PARAMETRIZE{ move = MOVE_CONFIDE; }
+    PARAMETRIZE{ move = MOVE_FAKE_TEARS; }
+    PARAMETRIZE{ move = MOVE_SCARY_FACE; }
+    PARAMETRIZE{ move = MOVE_SWEET_SCENT; }
+    PARAMETRIZE{ move = MOVE_SAND_ATTACK; }
+
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_GROWL].effect == EFFECT_ATTACK_DOWN);
+        ASSUME(gMovesInfo[MOVE_LEER].effect == EFFECT_DEFENSE_DOWN);
+        ASSUME(gMovesInfo[MOVE_CONFIDE].effect == EFFECT_SPECIAL_ATTACK_DOWN);
+        ASSUME(gMovesInfo[MOVE_FAKE_TEARS].effect == EFFECT_SPECIAL_DEFENSE_DOWN_2);
+        ASSUME(gMovesInfo[MOVE_SCARY_FACE].effect == EFFECT_SPEED_DOWN_2);
+        ASSUME(gMovesInfo[MOVE_SWEET_SCENT].effect == (B_UPDATED_MOVE_DATA >= GEN_6 ? EFFECT_EVASION_DOWN_2 : EFFECT_EVASION_DOWN));
+        ASSUME(gMovesInfo[MOVE_SAND_ATTACK].effect == EFFECT_ACCURACY_DOWN);
+        PLAYER(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_BELDUM) { Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN { MOVE(player, move); }
+    } SCENE {
+        NONE_OF {
+            ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent);
+        }
+        ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        MESSAGE("Foe Beldum's Clear Body prevents stat loss!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body prevents Sticky Web")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_STICKY_WEB].effect == EFFECT_STICKY_WEB);
+        PLAYER(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_BELDUM) { Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_STICKY_WEB); }
+        TURN { SWITCH(opponent, 1); }
+    } SCENE {
+        NONE_OF {
+            ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent);
+        }
+        ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+        MESSAGE("Foe Beldum's Clear Body prevents stat loss!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent stat stage reduction from moves used by the user")
+{
+    GIVEN {
+        ASSUME(MoveHasAdditionalEffectSelf(MOVE_SUPERPOWER, MOVE_EFFECT_ATK_DEF_DOWN) == TRUE);
+        PLAYER(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_BELDUM) { Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SUPERPOWER); }
+    } SCENE {
+        NONE_OF {
+            ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY); 
+            MESSAGE("Foe Beldum's Clear Body prevents stat loss!");
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Mold Breaker, Teravolt, and Turboblaze ignore Clear Body")
```

Comment:
I've watched the recording of the tests and it uses Armaldo and Kingler in place of Beldum. Could anybody confirm?

Reply:
I haven't been able to reproduce this, my test.elf is still showing Beldum. Is it just for this specific test or for all of them?

Reply:
Okay cool, no worries! Glad it got figured out :)